### ISSUE TITLE
upper_case_acronyms should not suggest breaking changes to APIs

### ISSUE THREAD

**Issue by sunshowers** [rust-lang/rust-clippy#6974]
Hi there! Thanks for writing `clippy` and all the useful suggestions it provides :)

Unfortunately, there's one lint that I've not been happy about: this new `upper_case_acronyms` lint that's on by default in Rust 1.51. I'm here to complain about two separate issues:
(1) I have aesthetic issues with the suggestions it makes.
(2) The more fundamental issue -- it routinely suggests breaking changes to stable, public APIs.

For example, the code here:

https://github.com/withoutboats/camino/blob/8f8d61dc91051b8a2ed013eb2f1fc34c03403bdb/src/lib.rs#L1595-L1629

has enums that are correctly and appropriately named, and [match the corresponding names in `std`](https://doc.rust-lang.org/std/path/enum.Prefix.html). However, the lint complains with:

![image](https://user-images.githubusercontent.com/180618/112557717-553d0e00-8d8a-11eb-8da9-f95bf68ec0e5.png)

**Issue 1**: the changes it suggests are aesthetically bad. `UNC` in particular is widely understood in the Windows world, and renaming it to `Unc` looks ugly and pattern-matches in my head to "Uncle".

**Issue 2**: Even if you disagree about the aesthetics, the lint recommends breaking changes to a public API. A lint like this couldn't possibly be high-value enough to suggest a breaking change to an API.

Please disable this lint by default, or at least for public APIs.

### Meta

- `cargo clippy -V`: clippy 0.1.51 (2fd73fa 2021-03-23)
- `rustc -Vv`:
```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-unknown-linux-gnu
release: 1.51.0
LLVM version: 11.0.1
```

**Comment by giraffate** [comment]
Thanks for your report!

This seems to be fixed by https://github.com/rust-lang/rust-clippy/pull/6805.

**Comment by sunshowers** [comment]
Thanks! Looks like that's not in the version of clippy released with Rust 1.51.0. Are there plans to do a patch release with this fix?

**Comment by giraffate** [comment]
Hmm, looking at the PR and issues, it doesn't seem to be discussed so far.

**Comment by sunshowers** [comment]
I really would suggest considering a patch release.

**Comment by matthiaskrgr** [comment]
Thanks for bringing this to our attention!

````rust
pub enum Wasd {
    SomeNAME,
    ANotherNAME,
}
````
This ONLY lints on stable right now (beta is already fixed, yay)

There are other things already on the point-release pile, maybe we can just sneak this one in...? :sweat_smile: 
I'm not sure what the release team thinks about this....
clippy topic thread
https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/upper_case_acronyms.20FP.20for.20pub.20items.20on.20stable

point release thread
https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/1.2E51.2E1.20point.20release

**Comment by flip1995** [comment]
Since Thursday is the new release and this is fixed in the current beta, I'll close this. Obviously there won't be a point release until Thursday.
